### ISSUE TITLE
memory safety bug with enqueueMap*

### ISSUE THREAD

**Issue by andreasgal** [mikeseven/node-opencl#43]
mPtr is assigned to the pointer returned by enqueue and it is then wrapped into a buffer. The buffer is returned. If the user does an enqueueUnmap*, mPtr is freed, but the buffer object might still be around, and accessing that is bad news.

To make matters worse, ArrayBuffers can be sliced and TypedArray views can be wrapped around them, and all those instances become unsafe.

No trivial fix for this, except copying, which would suck.

**Comment by mikeseven** [comment]
Yes I know, it's unsafe but I didn't want to do a copy. Maybe there is a way to detect such bad conditions.

--mike



On Wed, Apr 6, 2016 at 4:42 PM -0700, "Andreas Gal" <notifications@github.com<mailto:notifications@github.com>> wrote:


mPtr is assigned to the pointer returned by enqueue and it is then wrapped into a buffer. The buffer is returned. If the user does an enqueueUnmap*, mPtr is freed, but the buffer object might still be around, and accessing that is bad news.

To make matters worse, ArrayBuffers can be sliced and TypedArray views can be wrapped around them, and all those instances become unsafe.

No trivial fix for this, except copying, which would suck.

-
